vsphere/resource_vsphere_host.go

```diff
@@ -0,0 +1,683 @@
+	"github.com/vmware/govmomi/license"
+
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/clustercomputeresource"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceVsphereHost() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVsphereHostCreate,
+		Read:   resourceVsphereHostRead,
+		Update: resourceVsphereHostUpdate,
+		Delete: resourceVsphereHostDelete,
+		// Importer: ,
+		Schema: map[string]*schema.Schema{
+			"cluster": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the vSphere cluster the host will belong to.",
+			},
+			"hostname": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "FQDN or IP address of the host.",
+			},
+			"username": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Username of the administration account of the host.",
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Password of the administration account of the host.",
+				Sensitive:   true,
+			},
+			"thumbprint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Host's certificate SHA-1 thumbprint. If not set then the CA that signed the host's certificate must be trusted.",
+			},
+			"license": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "License key that will be applied to this host.",
+			},
+			"force": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Force add the host to vsphere, even if it's already managed by a different vSphere instance.",
+				Default:     false,
+			},
+			"connected": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Set the state of the host. If set to false then the host will be asked to disconnect.",
+				Default:     true,
+			},
+			"maintenance": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Set the host's maintenance mode. Default is false",
+				Default:     false,
+			},
+			"lockdown": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Set the host's lockdown status. Default is disabled. Valid options are 'disabled', 'normal', 'strict'",
+				Default:     "disabled",
+			},
+		},
+	}
+}
+
+func resourceVsphereHostRead(d *schema.ResourceData, meta interface{}) error {
+
+	// NOTE: Destroying the host without telling vsphere about it will result in us not
+	// knowing that the host does not exist any more.
+
+	// Look for host
+	client := meta.(*VSphereClient).vimClient
+	hostID := d.Id()
+	if hostID == "" {
```

Comment:
It should be safe to skip checking this. If the ID is empty, then the resource shouldn't exist.

Reply:
👍